### PR TITLE
[CDAP-15416] Migrates json-editor from angular to react in plugin configuration

### PR DESCRIPTION
**Pull request by ajainarayanan**
JIRA: https://issues.cask.co/browse/CDAP-15416
Build: https://builds.cask.co/browse/CDAP-UDUT313

### CODE REVIEW

File: cdap-ui/app/cdap/components/CodeEditor/JSONEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      activeLineMarker={false}
+      showPrettyPrintButton={true}
+      prettyPrintFunction={(unformattedValue: string) => {
+        let v = unformattedValue;
+        try {
+          v = JSON.parse(v);
+        } catch (e) {
+          return unformattedValue;
+        }
+        return JSON.stringify(v, null, 2);
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/CodeEditor/JSONEditor.tsx]: if `v` is a non-json string, this will cause double quotation

Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/CodeEditor/JSONEditor.tsx]: I am not sure if I follow. The `JSON.parse` and the `return` inside the `catch` block are in place to make sure `v` is `JSON`. Can you give an example of when doing JSON.stringify will cause an issue?

Comment (elfenheart) [reply cdap-ui/app/cdap/components/CodeEditor/JSONEditor.tsx]: nevermind, please disregard. Completely missed the return statement on the catch block.

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Comment by ajainarayanan** [comment]
Latest green build: https://builds.cask.co/browse/CDAP-UDUT313-4

**Pull request merged**